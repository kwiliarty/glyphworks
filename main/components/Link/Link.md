`Link` is a pretty much just a link. 

```jsx
<p>
  Here is a link to <Link href='https://glyph.works'>GlyphWorks</Link>
</p>
```

It can take a custom color:

```jsx
<p>
  See the <Link href='https://glyph.works' color='darkRed'>Custom Color</Link>
</p>
```
