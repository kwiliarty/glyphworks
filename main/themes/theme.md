## Settings

```jsx noeditor
import theme from './glyphworks'

;
<dl>
  {
    Object.keys( theme ).map( key => {
      if ( typeof( theme[ key ] ) == 'string' ) {
        return (
          <React.Fragment key={ key } >
            <dt><strong>{ key }</strong></dt>
            <dd><pre>{ theme[ key ] }</pre></dd>
          </React.Fragment>
        )
      }
    })
  }
</dl>
```
