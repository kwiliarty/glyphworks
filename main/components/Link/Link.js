import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import theme from '../../themes/glyphworks'

const StyledLink = styled.a`
  display: inline-block;
  color: ${ theme.colors.lightPurple };
  &:focus {
    outline: 2px dotted;
  }
  &:hover {
    outline: 1px dotted;
  }
`

const Link = props => {
  return (
    <StyledLink { ...props }>
      { props.children }
    </StyledLink>
  )
}

Link.propTypes = {
  /** This component renders its children. */
  children: PropTypes.node,
}

export default Link
