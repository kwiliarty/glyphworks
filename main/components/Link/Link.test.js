import React from 'react'
import { render, screen } from 'utils'
import Link from './Link.js'
import theme from 'glyphworks'

describe( 'The Link Component', () => {
  it( 'renders an anchor tag', () => {
    render( <Link href='https://glyph.works'>GlyphWorks</Link> )
    const link = screen.getByRole( 'link' )
    expect( link ).toHaveTextContent( 'GlyphWorks' )
    expect( link ).toHaveAttribute( 'href', 'https://glyph.works' )
    expect( link ).toMatchSnapshot()
  })

  it( 'accepts a color attribute', () => {
    render(
      <Link
        href='https://glyph.works'
        color='darkRed'
      >
        Custom color
      </Link>
    )
    const link = screen.getByRole( 'link' )
    expect( link ).toHaveTextContent( 'Custom' )
    expect( link ).toHaveStyleRule('color', theme.colors.darkRed )
    expect( link ).toMatchSnapshot()
  })
})
