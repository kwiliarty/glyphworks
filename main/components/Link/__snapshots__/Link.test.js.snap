// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Link Component accepts a color attribute 1`] = `
.c0 {
  display: inline-block;
  color: #621818;
}

.c0:focus {
  outline: 2px dotted;
}

.c0:hover {
  outline: 1px dotted;
}

<a
  class="c0"
  color="darkRed"
  href="https://glyph.works"
>
  Custom color
</a>
`;

exports[`The Link Component renders an anchor tag 1`] = `
.c0 {
  display: inline-block;
  color: #635a8a;
}

.c0:focus {
  outline: 2px dotted;
}

.c0:hover {
  outline: 1px dotted;
}

<a
  class="c0"
  color="lightPurple"
  href="https://glyph.works"
>
  GlyphWorks
</a>
`;
